worker/dblogpruner/worker.go

```diff
@@ -50,15 +48,48 @@ type pruneWorker struct {
 }
 
 func (w *pruneWorker) loop(stopCh <-chan struct{}) error {
+
+	controllerConfigWatcher := w.st.WatchControllerConfig()
+	defer worker.Stop(controllerConfigWatcher)
+
+	var (
+		maxLogAge               time.Duration
+		maxCollectionMB         int
+		controllerConfigChanges = controllerConfigWatcher.Changes()
+		// We will also get an initial event, but need to ensure that event is
+		// received before doing any pruning.
+		haveConfig = false
+	)
 	p := w.params
+
 	for {
 		select {
 		case <-stopCh:
 			return tomb.ErrDying
+		case _, ok := <-controllerConfigChanges:
+			if !ok {
+				return errors.New("controller configuration watcher closed")
+			}
+			controllerConfig, err := w.st.ControllerConfig()
+			if err != nil {
+				return errors.Annotate(err, "cannot load controller configuration")
+			}
+			haveConfig = true
+			newMaxAge := controllerConfig.MaxLogsAge()
+			newMaxCollectionMB := controllerConfig.MaxLogSizeMB()
+			if newMaxAge != maxLogAge || newMaxCollectionMB != maxCollectionMB {
+				logger.Infof("log pruning config: max age: %v, max collection size %dM", newMaxAge, newMaxCollectionMB)
+				maxLogAge = newMaxAge
+				maxCollectionMB = newMaxCollectionMB
+			}
+			continue
```

Comment:
it also forces the check for a new update prior to doing the pruning in case one comes in at the same time